recipe/install-sysroot.sh

```diff
@@ -0,0 +1,234 @@
+  # fi
+
+  make prefix=${PREFIX} install-libcc1
+  install -d ${PREFIX}/share/gdb/auto-load/usr/lib
+
+  make prefix=${PREFIX} install-fixincludes
+  make -C gcc prefix=${PREFIX} install-mkheaders
+
+  if [[ -d ${CHOST}/libgomp ]]; then
+    make -C ${CHOST}/libgomp prefix=${PREFIX} install-nodist_{libsubinclude,toolexeclib}HEADERS
+  fi
+
+  if [[ -d ${CHOST}/libitm ]]; then
+    make -C ${CHOST}/libitm prefix=${PREFIX} install-nodist_toolexeclibHEADERS
+  fi
+
+  if [[ -d ${CHOST}/libquadmath ]]; then
+    make -C ${CHOST}/libquadmath prefix=${PREFIX} install-nodist_libsubincludeHEADERS
+  fi
+
+  if [[ -d ${CHOST}/libsanitizer ]]; then
+    make -C ${CHOST}/libsanitizer prefix=${PREFIX} install-nodist_{saninclude,toolexeclib}HEADERS
+  fi
+
+  if [[ -d ${CHOST}/libsanitizer/asan ]]; then
+    make -C ${CHOST}/libsanitizer/asan prefix=${PREFIX} install-nodist_toolexeclibHEADERS
+  fi
+
+  if [[ -d ${CHOST}/libsanitizer/tsan ]]; then
+    make -C ${CHOST}/libsanitizer/tsan prefix=${PREFIX} install-nodist_toolexeclibHEADERS
+  fi
+
+  make -C libiberty prefix=${PREFIX} install
+  # install PIC version of libiberty
+  install -m644 libiberty/pic/libiberty.a ${PREFIX}/lib
+
+  make -C gcc prefix=${PREFIX} install-man install-info
+
+  make -C gcc prefix=${PREFIX} install-po
+
+  # many packages expect this symlink
+  [[ -f ${PREFIX}/bin/${CHOST}-cc ]] && rm ${PREFIX}/bin/${CHOST}-cc
+  pushd ${PREFIX}/bin
+    ln -s ${CHOST}-gcc ${CHOST}-cc
+  popd
+
+  # POSIX conformance launcher scripts for c89 and c99
+  cat > ${PREFIX}/bin/c89 <<"EOF"
+#!/bin/sh
+fl="-std=c89"
+for opt; do
+  case "$opt" in
+    -ansi|-std=c89|-std=iso9899:1990) fl="";;
+    -std=*) echo "`basename $0` called with non ANSI/ISO C option $opt" >&2
+      exit 1;;
+  esac
+done
+exec gcc $fl ${1+"$@"}
+EOF
+
+  cat > ${PREFIX}/bin/c99 <<"EOF"
+#!/bin/sh
+fl="-std=c99"
+for opt; do
+  case "$opt" in
+    -std=c99|-std=iso9899:1999) fl="";;
+    -std=*) echo "`basename $0` called with non ISO C99 option $opt" >&2
+      exit 1;;
+  esac
+done
+exec gcc $fl ${1+"$@"}
+EOF
+
+  chmod 755 ${PREFIX}/bin/c{8,9}9
+
+  rm ${PREFIX}/bin/${CHOST}-gcc-${PKG_VERSION}
+
+popd
+
+# Install kernel headers
+kernel_arch=${ctng_cpu_arch}
+if [[ ${kernel_arch} == aarch64 ]]; then
+  kernel_arch=arm64
+elif [[ ${kernel_arch} == i686 ]]; then
+  kernel_arch=x86
+fi
+
+make -C ${SRC_DIR}/.build/src/linux-* CROSS_COMPILE=${CHOST}- O=${SRC_DIR}/.build/${CHOST}/build/build-kernel-headers ARCH=${kernel_arch} INSTALL_HDR_PATH=${PREFIX}/${CHOST}/sysroot/usr ${VERBOSE_AT} headers_install
+
+if [[ ${ctng_libc} == gnu ]]; then
+  # Install libc libraries
+  pushd ${SRC_DIR}/.build/${CHOST}/build/build-libc-final/multilib
+    make -l BUILD_CFLAGS="-O2 -g -I${SRC_DIR}/.build/${CHOST}/buildtools/include" \
+            BUILD_LDFLAGS="-L${SRC_DIR}/.build/${CHOST}/buildtools/lib"           \
+            install_root=${PREFIX}/${CHOST}/sysroot install
```

Comment:
How about just not doing this install in `install-gcc.sh` and then adding the CDT based `sysroot_linux-64` as a run requirement?

Reply:
happy to try!

I don’t follow exactly which install commands in these scripts do what.

Reply:
It looks like there is another install into the sysroot below and maybe a bit above. I think we’ll have to build it a few times to check then outputs to make sure it is split properly.